cwms-data-api/src/main/java/cwms/cda/data/dto/watersupply/WaterUserContract.java

```diff
@@ -0,0 +1,241 @@
+/*
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Hydrologic Engineering Center
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package cwms.cda.data.dto.watersupply;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import cwms.cda.data.dto.CwmsDTO;
+import cwms.cda.data.dto.CwmsId;
+import cwms.cda.data.dto.LookupType;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.annotations.FormattableWith;
+import cwms.cda.formatters.json.JsonV1;
+import java.util.Date;
+
+@FormattableWith(contentType = Formats.JSONV1, formatter = JsonV1.class)
+@JsonDeserialize(builder = WaterUserContract.Builder.class)
+public class WaterUserContract extends CwmsDTO {
+
+    @JsonProperty(required = true)
+    private final WaterUser waterUser;
+    @JsonProperty(required = true)
+    private final CwmsId contractId;
+    @JsonProperty(required = true)
+    private final LookupType waterContract;
+    @JsonProperty(required = true)
+    private final Date contractEffectiveDate;
+    @JsonProperty(required = true)
+    private final Date contractExpirationDate;
+    @JsonProperty(required = true)
+    private final Double contractedStorage;
+    @JsonProperty(required = true)
+    private final Double initialUseAllocation;
+    @JsonProperty(required = true)
+    private final Double futureUseAllocation;
+    @JsonProperty(required = true)
+    private final String storageUnitsId;
+    @JsonProperty(required = true)
+    private final Double futureUsePercentActivated;
+    @JsonProperty(required = true)
+    private final Double totalAllocPercentActivated;
+    private final WaterSupplyPump pumpOutLocation;
```

Comment:
The generated JSON for these has redundancies. I'm not sure if that is a big deal though.
  "pump-out-location": {
    "pump-location": {
      "office-id": "MVR",
      "name": "PUMP1",
      "latitude": 0,
      "longitude": 0,
      "active": true,
      "public-name": "Test Public Pump Name",
      "long-name": "Test Long Name",
      "description": "Test Description",
      "timezone-name": "UTC",
      "location-type": "Test Location Type",
      "location-kind": "PUMP",
      "nation": "US",
      "state-initial": "NV",
      "county-name": "Clark",
      "nearest-city": "Sparks",
      "horizontal-datum": "WGS84",
      "published-longitude": 0,
      "published-latitude": 0,
      "vertical-datum": "WGS84",
      "elevation": 150,
      "map-label": "Test Map Label",
      "bounding-office-id": "MVR",
      "elevation-units": "m"
    },
    "pump-type": "PUMP_OUT"
  },

Reply:
I tried to reduce redundancies, but there are still duplicate office IDs. Not sure if it makes a big difference

Reply:
duplicate offices are expected